tiledb/api/src/array/dimension/mod.rs

```diff
@@ -345,6 +371,21 @@ impl DimensionConstraints {
             std::ptr::null()
         )
     }
+
+    /// Returns the number of cells spanned by this constraint, if applicable
+    pub fn num_cells(&self) -> Option<u128> {
```

Comment:
Ought to be enough for proptest, at any rate.  I was getting panics on overflows when proptest gave me a dimension ranging from something close to `[u64::MIN, u64::MAX]`.  `usize` is not large enough to represent that, we need to hold `u64::MAX * 2`.

...

Well that doesn't really seem right, does it... maybe using `i128` as a temporary should be good enough, and we can return `u64` here.  Though it is useful to keep `u128` because we will, for various reasons, want to compute a product on the total number of cells in the whole schema, and it's easier to check the result at the end than it is to check for overflow during the steps of the product.

Reply:
Ah, products is what I was missing.